### DISCUSSION TITLE
自定义Deepl的api接口

### DISCUSSION THREAD

**Discussion by congnews** [immersive-translate/immersive-translate#994] [Ideas]
### 新功能 | New Feature

deepl-free的接口访问不是很稳定、自己反代了接口，但是不能自定义接口地址。

### 为什么您想要添加这些功能 | Why you want to add these features

可以自定义Deepl的api接口

### 提议 | Proposal

自定义Deepl的api接口

**Comment by theowenyoung** [comment]
在插件设置里开启开发者设置，会出现一个 deeplx 的翻译服务，请在这里自定义 url 即可。

**Comment by congnews** [comment]
那deepl key在哪填？

**Reply by theowenyoung** [reply]
接口规范是按照这个： https://github.com/OwO-Network/DeepLX#use-in-python

如果需要key的话，你可以在url 上自己实现一个？ api_key=xxxxxxx

**Reply by congnews** [reply]
可以直接像Openai一样支持自定义api地址吗？这样使用起来更直接更方便。

**Reply by theowenyoung** [reply]
deepl 自定义地址是很小众的需求，有能力搭建自己自定义接口的同学，目前的方式理论上也很简单～
